src/cpl/nuopc/rof_comp_nuopc.F90

```diff
@@ -0,0 +1,886 @@
+    allocate(gindex(lsize))
+    ni = 0
+    do n = runoff%begr,runoff%endr
+       ni = ni + 1
+       gindex(ni) = runoff%gindex(n)
+    end do
+
+    ! create distGrid from global index array
+    DistGrid = ESMF_DistGridCreate(arbSeqIndexList=gindex, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    deallocate(gindex)
+
+    ! read in the mesh
+    call NUOPC_CompAttributeGet(gcomp, name='mesh_rof', value=cvalue, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    EMesh = ESMF_MeshCreate(filename=trim(cvalue), fileformat=ESMF_FILEFORMAT_ESMFMESH, &
+          elementDistgrid=Distgrid, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    if (masterproc) then
+       write(iulog,*)'mesh file for domain is ',trim(cvalue)
+    end if
+
+    !--------------------------------
+    ! realize actively coupled fields
+    !--------------------------------
+
+    call realize_fields(gcomp,  Emesh, flds_scalar_name, flds_scalar_num, rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    !--------------------------------
+    ! Create RTM export state
+    !--------------------------------
+
+    call export_fields(gcomp, rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    ! Set global grid size scalars in export state
+    call State_SetScalar(dble(rtmlon), flds_scalar_index_nx, exportState, &
+         flds_scalar_name, flds_scalar_num, rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    call State_SetScalar(dble(rtmlat), flds_scalar_index_ny, exportState, &
+         flds_scalar_name, flds_scalar_num, rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    !----------------------------------------------------------------------------
+    ! Reset shr logging
+    !----------------------------------------------------------------------------
+
+    call shr_file_setLogUnit (shrlogunit)
+
+    call ESMF_LogWrite(subname//' done', ESMF_LOGMSG_INFO)
+
+    !--------------------------------
+    ! diagnostics
+    !--------------------------------
+
+    if (debug > 1) then
+       call State_diagnose(exportState,subname//':ES',rc=rc)
+       if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    endif
+
+#ifdef USE_ESMF_METADATA
+    convCIM  = "CIM"
+    purpComp = "Model Component Simulation Description"
+    call ESMF_AttributeAdd(comp, convention=convCIM, purpose=purpComp, rc=rc)
+    call ESMF_AttributeSet(comp, "ShortName", "RTM", convention=convCIM, purpose=purpComp, rc=rc)
+    call ESMF_AttributeSet(comp, "LongName", "RTM River Model", convention=convCIM, purpose=purpComp, rc=rc)
+    call ESMF_AttributeSet(comp, "Description", "RTM River Model", convention=convCIM, purpose=purpComp, rc=rc)
+    call ESMF_AttributeSet(comp, "ReleaseDate", "2017", convention=convCIM, purpose=purpComp, rc=rc)
+    call ESMF_AttributeSet(comp, "ModelType", "River", convention=convCIM, purpose=purpComp, rc=rc)
+    call ESMF_AttributeSet(comp, "Name", "TBD", convention=convCIM, purpose=purpComp, rc=rc)
```

Comment:
I think this name and contact should be ctsm_software@ucar.edu

Reply:
see above.